### PR TITLE
Popup window to be centred on screen

### PR DESCRIPTION
**Pull request by thejurassic**
## What does this pull request change?
Popup window to be centred in the window (or screen)

## Why is this pull request needed?
Popup window is not centred, and would provide better user experience if centred on the page theyre working on that invoked the login

## Issues related to this change
None

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -1,133 +1,154 @@
...
   TTokenRequestForRefresh,
   TTokenRequestWithCodeAndVerifier,
   TTokenResponse,
-} from './types'
+} from "./types";
 
-const codeVerifierStorageKey = 'PKCE_code_verifier'
-const stateStorageKey = 'ROCP_auth_state'
+const codeVerifierStorageKey = "PKCE_code_verifier";
+const stateStorageKey = "ROCP_auth_state";
```
Comment (sebastianvitterso) [inline src/authentication.ts:14]: I can tell you right now that you should try to run the installed pre-commit hooks, as all these changes go against the style guide in the repo 😄 

We'll look into the changes when the changes aren't all over!

**Review by sebastianvitterso — CHANGES_REQUESTED** [review]

**Comment by sebastianvitterso** [comment]
@thejurassic, are you still interested in this change?

File: src/authentication.ts [hunk 2]
```diff
@@ -54,14 +55,45 @@ export async function redirectToLogin(
...
+
+  left = Math.min(left, maxLeft) // Ensure the right side is within the screen
+  top = Math.min(top, maxTop) // Ensure the bottom side is within the screen
+
+  // Ensure the popup is not positioned off the top/left of the screen
+  left = Math.max(0, left)
+  top = Math.max(0, top)
+
+  return { left, top, width: popupWidth, height: popupHeight }
+}
```
Comment (sebastianvitterso) [inline src/authentication.ts:95]: This should probably be moved to a separate utility-file, as it doesn't belong in `authentication.ts`, in our opinion.

File: src/authentication.ts [hunk 3]
```diff
@@ -54,14 +55,45 @@ export async function redirectToLogin(
...
     window.location.assign(loginUrl)
   })
 }
 
+function calculatePopupPosition(popupWidth = 600, popupHeight = 600): TPopupPosition {
+  // Calculate the screen dimensions and position the popup at the center
+  const screenLeft = window.screenLeft || window.screenX
+  const screenTop = window.screenTop || window.screenY
+  const screenWidth = window.innerWidth || document.documentElement.clientWidth || screen.width
+  const screenHeight = window.innerHeight || document.documentElement.clientHeight || screen.height
```
Comment (sebastianvitterso) [inline src/authentication.ts:76]: Will any of these actually be unset at any point? 

E.g. `screenLeft` vs `screenX` has basically the same stats on caniuse ([screenLeft](https://caniuse.com/mdn-api_window_screenleft), [screenX](https://caniuse.com/mdn-api_window_screenx)), and I assume the same may be true for the other 6 properties you've listed.

If so, let's just use the most "natural"/simple ones, and stick with those.

File: src/authentication.ts [hunk 4]
```diff
@@ -54,14 +55,45 @@ export async function redirectToLogin(
...
+  const maxTop = screenTop + (screenHeight - popupHeight)
+
+  left = Math.min(left, maxLeft) // Ensure the right side is within the screen
+  top = Math.min(top, maxTop) // Ensure the bottom side is within the screen
+
+  // Ensure the popup is not positioned off the top/left of the screen
+  left = Math.max(0, left)
+  top = Math.max(0, top)
+
+  return { left, top, width: popupWidth, height: popupHeight }
```
Comment (sebastianvitterso) [inline src/authentication.ts:94]: This is cool and all, but if this is meant to fix all cases, what about the case where the screen is smaller than the dialog size? Then surely, we should resize the dialog to match the screen?

So

```js
const maxLeft = screenLeft + (screenWidth - popupWidth)
const maxTop = screenTop + (screenHeight - popupHeight)

return {
  left: Math.max(0, Math.min(left, maxLeft)),
  top: Math.max(0, Math.min(top, maxTop)),
  width: Math.min(screenWidth, popupWidth),
  height: Math.min(screenHeight, popupHeight),
}
```

File: src/authentication.ts (hunk 3 above)
Comment (sebastianvitterso) [reply src/authentication.ts:76]: And as such, if these are set in (basically) all browsers, consider using them directly further down into the script.

File: src/authentication.ts (hunk 4 above)
Comment (sebastianvitterso) [reply src/authentication.ts]: Did this myself  ✌️

File: src/authentication.ts (hunk 2 above)
Comment (sebastianvitterso) [reply src/authentication.ts]: Did this myself  ✌️

**Review by soofstad — APPROVED** [review]
Looks good! Tested OK.
Thank you for the contribution @thejurassic :tada:

**Pull request merged**